### PR TITLE
Change gen_openapi.sh to generate schema with generic mount paths

### PR DESCRIPTION
**Pull request by averche**
`gen_openapi.sh` is a script used to generate `OpenAPI` schema for all built-in plugins. The resultant `openapi.json` schema is then used for client library generation.

This pull request fixes a couple issues:
- The script will now ignore plugins marked as "Removed" (these were causing script failures)
- The will now generate schema with `generic_mount_paths` set to true, inserting `/{thing_mount_path}/` elements into the paths for certain plugins. This was the default behavior before https://github.com/hashicorp/vault/pull/18617. We are now re-enabling the behavior but specifically for the schemas generated using this script.

### CODE REVIEW

File: scripts/gen_openapi.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ then
     exit 1
 fi
 
-vault server -dev -dev-root-token-id=root &
+../bin/vault server -dev -dev-root-token-id=root &
```
Comment (AnPucel) [inline scripts/gen_openapi.sh:23]: Is this for local testing?

Comment (averche) [reply scripts/gen_openapi.sh:23]: 🤦

File: scripts/gen_openapi.sh [hunk 2]
```diff
@@ -36,7 +36,7 @@ while read -r line; do
         codeLinesStarted=true
     elif [[ $line == *"databasePlugins:"* ]] ; then
         break
-    elif [ $codeLinesStarted = true ] && [[ $line == *"consts.Deprecated"* || $line == *"consts.PendingRemoval"* ]] ; then
+    elif [ $codeLinesStarted = true ] && [[ $line == *"consts.Deprecated"* || $line == *"consts.PendingRemoval"* || $line == *"consts.Removed"* ]] ; then
```
Comment (AnPucel) [inline scripts/gen_openapi.sh:39]: Is this based off the comments in the registry like we previously had it? Or is this something new?

Comment (AnPucel) [reply scripts/gen_openapi.sh:39]: Just out of curiosity b/c I don't remember anything that had said `PendingRemoval` or `Removed`

Comment (averche) [reply scripts/gen_openapi.sh:39]: They are no longer comments but actual struct values:

https://github.com/hashicorp/vault/blob/fd9cadb192e37c398f56c7fffe7f82832e166790/helper/builtinplugins/registry.go#L100-L103

What we are doing here (parsing Go code in shell) is kind of gross. I'm just fixing the script while we are still using it.

Comment (AnPucel) [reply scripts/gen_openapi.sh:39]: oh okay! gotcha. At least the deprecation status is codified in here though.

**Review by AnPucel — APPROVED** [review]

**Pull request merged**